tests/e2e_tests/flatfield_e2e.py

```diff
@@ -0,0 +1,368 @@
+
+
+    ####### Test the flat field result
+    flat_filename = l1_flatfield_filelist[0].split(os.path.sep)[-1].replace("_L1_", "_L2a_")[:-5] + "_flatfield.fits"
+    flat = data.FlatField(os.path.join(flat_outputdir, flat_filename))
+    good_region = np.where(flat.data != 1)
+    diff = flat.data - input_flat # compute residual from true
+    smoothed_diff = scipy.ndimage.gaussian_filter(diff, 1.4) # smooth by the size of the resolution element, since we care about that
+    print(np.std(smoothed_diff[good_region]))
+    assert np.std(smoothed_diff[good_region]) < 0.0071
+
+
+    ####### Check the bad pixel map result
+    bp_map_filename = "mock_detnoisemaps_dark_bad_pixel_map.fits"
+    bpmap = data.BadPixelMap(os.path.join(flat_outputdir, bp_map_filename))
+    assert np.all(bpmap.data == 0) # this bpmap should have no bad pixels
+
+    # clean up by removing entry
+    this_caldb.remove_entry(nonlinear_cal)
+    this_caldb.remove_entry(kgain)
+    this_caldb.remove_entry(noise_map)
+    this_caldb.remove_entry(flat)
+    this_caldb.remove_entry(bpmap)
+
+
+@pytest.mark.e2e
+def test_flat_creation_uranus(tvacdata_path, e2eoutput_path):
+    """
+    Tests e2e flat field using Uranus in Band 1, only HLC FOV
+
+    Args:
+        tvacdata_path (str): path to CGI_TVAC_Data dir
+        e2eoutput_path (str): output directory
+    """
+    # figure out paths, assuming everything is located in the same relative location
+    l1_dark_datadir = os.path.join(tvacdata_path, "TV-20_EXCAM_noise_characterization", "darkmap")
+    processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
+
+    # make output directory if needed
+    flat_outputdir = os.path.join(e2eoutput_path, "flat_uranus_output")
+    if not os.path.exists(flat_outputdir):
+        os.mkdir(flat_outputdir)
+    flat_mock_inputdir = os.path.join(flat_outputdir, "mock_input_data")
+    if not os.path.exists(flat_mock_inputdir):
+        os.mkdir(flat_mock_inputdir)    
+
+    # assume all cals are in the same directory
+    nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+    dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")
+    flat_path = os.path.join(processed_cal_path, "flat.fits")
+    fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+    cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+    bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+    # mock flat field is all ones
+    input_flat = np.ones([1024, 1024], dtype=float)
+    # input_flat = np.random.normal(1,.03,(1024, 1024))
+    # create mock onsky rasters
+    hstdata_filedir = os.path.join(thisfile_dir,"..", "test_data")
+    hstdata_filenames = glob.glob(os.path.join(hstdata_filedir, "med*.fits"))
+    hstdata_dataset = data.Dataset(hstdata_filenames)
+    raster_dataset = mocks.create_onsky_rasterscans(hstdata_dataset, planet='uranus',band='1',im_size=1024, d=65, n_dith=1, 
```

Comment:
doesn't Uranus need n_dith=2?

Reply:
I think I determined n=1 is more similar (but not exactly the same) as what was proposed in the Maier paper. They are not the same, but I also think that our exact dithering pattern is subject to obs. planning, so I didn't want to try to modify the code to get it to reproduce exactly what was done in Maier (it would involve changing substantially the code).